src/main/scala/io/iohk/ethereum/crypto/EthereumIESEngine.scala

```diff
@@ -111,7 +101,7 @@ class EthereumIESEngine(kdf: Either[ConcatKDFBytesGenerator, MGF1BytesGeneratorE
         val derivedKey: ByteString = fillKDFunction(ECIESCoder.KeySize / 8 + ECIESCoder.KeySize / 8)
         val (firstPart, secondPart) = derivedKey.splitAt(ECIESCoder.KeySize / 8)
 
-        if (IV != null)
+        if (Option(IV).isDefined)
```

Comment:
I'd revert this commit. Is there any reason why we could expect `null` in this place? Summoning @adamsmo to explain.